pkg/providers/instancetype/instancetype.go

```diff
@@ -311,6 +335,52 @@ func (p *DefaultProvider) createOfferings(ctx context.Context, instanceType *ec2
 			}).Set(price)
 		}
 	}
+
+	log.FromContext(ctx).WithValues("instanceType", *instanceType.InstanceType, "capacityReservations", capacityReservations).V(0).Info("capacity reservations for instanceType")
+	for _, capacityReservation := range capacityReservations {
+		log.FromContext(ctx).WithValues("instanceType", *instanceType.InstanceType, "capacityReservation", capacityReservation.ID).V(0).Info("capacity reservation for instanceType")
+		if capacityReservation.InstanceType != *instanceType.InstanceType {
+			log.FromContext(ctx).WithValues("instanceType", *instanceType.InstanceType, "capacityReservation", capacityReservation.ID).V(0).Info("capacity reservation not applicatble for instanceType")
+			continue
+		}
+		log.FromContext(ctx).WithValues("instanceType", *instanceType.InstanceType, "capacityReservation", capacityReservation.ID).V(0).Info("capacity reservation applicatble for instanceType")
+		price := 0.0
+		isAvailable := capacityReservation.AvailableInstanceCount > 0
+		zone := capacityReservation.AvailabilityZone
+		capacityType := ec2.UsageClassTypeOnDemand
+
+		subnet, hasSubnet := lo.Find(subnets, func(s v1beta1.Subnet) bool {
+			return s.Zone == zone
+		})
+		available := isAvailable && instanceTypeZones.Has(zone) && hasSubnet
+		offering := cloudprovider.Offering{
+			Requirements: scheduling.NewRequirements(
+				scheduling.NewRequirement(corev1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, capacityType),
+				scheduling.NewRequirement(v1.LabelTopologyZone, v1.NodeSelectorOpIn, zone),
+			),
+			Price:     price,
+			Available: available,
+		}
+		if subnet.ZoneID != "" {
+			offering.Requirements.Add(scheduling.NewRequirement(v1beta1.LabelTopologyZoneID, v1.NodeSelectorOpIn, subnet.ZoneID))
+		}
+		// TODO: tvonhacht
+		offering.Requirements.Add(scheduling.NewRequirement(v1beta1.LabelCapactiyReservationID, v1.NodeSelectorOpIn, capacityReservation.ID))
+		log.FromContext(ctx).WithValues("instanceType", *instanceType.InstanceType, "capacityReservation", capacityReservation.ID, "offering", offering).V(0).Info("offering for capacity reservation and instanceType")
+		offerings = append(offerings, offering)
+		instanceTypeOfferingAvailable.With(prometheus.Labels{
```

Comment:
I wonder if we punt on metrics for capacity reservations for instance type availability/price right now. This gets trickier since there's now no longer a concept of "global" availability and price but now availability is dictated on a per NodePool-basis, since the EC2NodeClass dictates the offerings that are surfaced for each one. Maybe we need to add a NodePool dimension to make this more accurate, but this is going to increase the cardinality even more than we already have (and this is a pretty high cardinality metric to begin with)